src/wallet/rpcdump.cpp

```diff
@@ -74,19 +77,166 @@ std::string DecodeDumpString(const std::string &str) {
+        return false;
+
+    // Add sweep destination - fee
+    mtx.vout.push_back(CTxOut(total, GetScriptForDestination(destAddress.Get())));
+    mtx.vout[0].nValue -= 3*CWallet::GetMinimumFee(GetVirtualTransactionSize(mtx), DEFAULT_TX_CONFIRM_TARGET, mempool);
+
+    mtx.nLockTime = chainActive.Height();
+
+    // Add the private key to a keystore
+    CBasicKeyStore tempKeystore;
+    tempKeystore.AddKey(dirtyKey);
+
+    const CTransaction txConst(mtx);
+    const CKeyStore& keystoreConst = tempKeystore;
+    for (uint32_t i = 0; i < mtx.vin.size(); i++)
+    {
+        CTxIn& input = mtx.vin[i];
+        TransactionSignatureCreator creator(&keystoreConst, &txConst, i, mtx.vout[input.prevout.n].nValue);
+
+        SignatureData sigdata;
+        bool sigCreated = ProduceSignature(creator, vToSweep[i].tx->vout[input.prevout.n].scriptPubKey, sigdata);
+
+        if (sigCreated && i < mtx.vin.size()) {
+            mtx.vin[i].scriptSig = sigdata.scriptSig;
+        }
+    }
+
+    // Commit the transaction
+    const CTransaction commit(mtx);
+    CValidationState state;
+    bool fMissingInputs;
+    if (!AcceptToMemoryPool(mempool, state, commit, false, &fMissingInputs, false, maxTxFee)) {
+        if (state.IsInvalid()) {
+            throw JSONRPCError(RPC_TRANSACTION_REJECTED, strprintf("%i: %s", state.GetRejectCode(), state.GetRejectReason()));
+        } else {
+            if (fMissingInputs) {
+                throw JSONRPCError(RPC_TRANSACTION_ERROR, "Missing inputs");
+            }
+            throw JSONRPCError(RPC_TRANSACTION_ERROR, state.GetRejectReason());
+        }
+
+        LogPrintf("SweepPrivKey(): Error: Transaction not valid\n");
+        return false;
+    }
+
+    if (pwalletMain->GetBroadcastTransactions() && !g_connman)
+        throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Peer-to-peer functionality missing or disabled");
+
+    // Broadcast the transaction
+    CInv inv(MSG_TX, commit.GetHash());
+    g_connman->ForEachNode([&inv](CNode* pnode)
+    {
+        pnode->PushInventory(inv);
+    });
+
+    return true;
+}
+
+UniValue sweepkey(const UniValue& params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+
+    if (fHelp || params.size() != 2)
+        throw runtime_error(
+            "sweepkey \"bitcoinprivkey\" \"bitcoinaddress\"\n"
+            "\nSweep the privkey's outputs to the specified address.\n"
+            "\nArguments:\n"
+            "1. \"bitcoinprivkey\"   (string, required) The private key (see dumpprivkey)\n"
+            "2. \"bitcoinaddress\"   (bitcoinaddress, required) The destination address\n"
+            "\nExamples:\n"
+            "\nSweep a private key\n"
+            + HelpExampleCli("sweepkey", "\"mykey\" \"myaddress\"") +
+            "\nAs a JSON-RPC call\n"
+            + HelpExampleRpc("sweepkey", "\"mykey\" \"myaddress\"")
+        );
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    EnsureWalletIsUnlocked();
```

Comment:
I was making this with the intention that the private key is actually in the wallet, or being imported, in which case it will be in the wallet before the sweep.